src/main/kotlin/lifeCycle/LifeCycle.kt

```diff
@@ -0,0 +1,32 @@
+package lifeCycle
+
+class LifeCycle(state: LifeCycleState) {
+    private var lifeCycleListeners = mutableListOf<LifeCycleListener>()
+
+    var state: LifeCycleState = state
+        private set
+
+    fun changeState(state: LifeCycleState) {
+        this.state = state
+        lifeCycleListeners.forEach { it.onStateChange(state) }
+        if (state == LifeCycleState.DESTROYED)
+            lifeCycleListeners.clear()
+    }
+
+    fun isChangeAble(): Boolean {
+        return when (state) {
+            LifeCycleState.LIVE -> true
+            else -> false
+        }
+    }
+
+    fun getLifeCycleListeners() = lifeCycleListeners.toList()
```

Comment:
```suggestion
    fun getListeners() = lifeCycleListeners.toList()
```
생략 할 수 있는것은 생략하는것으로 가독성과 유지보수성을 높일 수 있습니다.
Lifecycle의 Listeners 이므로 `getListeners` 함수명이면 충분할듯합니다.